lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -444,16 +456,22 @@ private JourneyResponse buildReuseResponse(
         }
 
         // check the result of 6MFC and return the appropriate journey
-        if (configService.enabled(REPEAT_FRAUD_CHECK.getName()) && !hasCurrentFraudVc(vcs)) {
+        if (configService.enabled(REPEAT_FRAUD_CHECK) && !hasCurrentFraudVc(vcs)) {
             LOGGER.info(
                     LogHelper.buildLogMessage(
                             "Fraud VC found and expired, resetting identity for GPG45 evaluation."));
+            sessionCredentialsService.persistCredentials(
+                    allVcsExceptFraud(vcs), auditEventUser.getSessionId());
```

Comment:
In this scenario, I think we'll have already persisted _all_ the credentials in `checkForProfileMatch`?

Reply:
Ooo yeah. So we need to remove the fraud VC, rather than add everything except. Althogh that raises the question of if we should be allowing expired VC to contribute when calculating gpg45 scores?

Reply:
It's a good question... Strictly speaking we probably should exclude it (since it's no longer a valid fraud score), but it's quite convenient to allow it...
- We want to check if you meet a GPG45 profile with the expired fraud VC before allowing you to take a repeat fraud check journey
- We are testing this by getting the fraud CRI to generate an expired fraud VC as part of a journey